samples/net/tftp_client/src/tftp-client.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2023 Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_tftp_client_app
+#define LOG_LEVEL LOG_LEVEL_DBG
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/tftp.h>
+
+#define APP_BANNER "Run TFTP client"
+
+#define TFTP_PORT		69
+#define TFTP_SERVER_ADDR	CONFIG_NET_CONFIG_PEER_IPV4_ADDR
+#define TFTP_SAMPLE_DATA	"Lorem ipsum dolor sit amet, consectetur adipiscing elit"
+
+static struct tftpc client;
+
+static void tftp_event_callback(const struct tftp_evt *evt)
+{
+	switch (evt->type) {
+	case TFTP_EVT_DATA:
+		LOG_HEXDUMP_INF(evt->param.data.data_ptr,
+				evt->param.data.len,
+				"Received data: ");
+		break;
+	case TFTP_EVT_ERROR:
+		LOG_ERR("Error code %d msg: %s",
+			evt->param.error.code,
+			evt->param.error.msg);
+	default:
+		break;
+	}
+}
+
+static int tftp_init(const char *hostname)
+{
+	struct sockaddr  server;
+	struct zsock_addrinfo *res, hints = {0};
+	int ret;
+
+	/* Set IP address directly */
+	server.sa_family = AF_INET;
+	ret = net_addr_pton(AF_INET, hostname,
+				&((struct sockaddr_in *)&server)->sin_addr);
+	/* Try to resolve DNS */
+	if (ret < 0) {
+#if defined(CONFIG_DNS_RESOLVER)
```

Comment:
#ifdef here shouldn't be needed - in case `DNS_RESOLVER` is disabled, and we do not provide IP literal, the `getaddrinfo()` should simply fail.